en/docs/learn/design-api/publish-api/publish-an-api-with-artifact-synchronization.md

```diff
@@ -0,0 +1,158 @@
+
+1. When an API gets Published, Edited, or removed, the synapse artifacts corresponding to that API will be Stored or
+ updated  in the extension point. 
+2. Then an event will be sent to Traffic Manager(TM) using Event Notifiers with API Name, UUID, and the gateway label
+ for the API.
+3. Gateways are subscribed to the TM. Gateway will filter out the events by the Gateway label and APIs that have the
+ gateway's label will be sorted. 
+4. Then it will fetch the artifacts associated with the API from the storage (Database or Github) and load it to the
+ memory.
+
+
+There will be an extension in the publisher profile to store the synapse artifacts in a persistence storage. The default implementation uses the API Manager Database itself. Once the API is Published, Edited, or removed, an event will be sent to Traffic Manager using Event Notifiers with API Name, UUID, and the gateway label for the API. 
+
+
+##API Gateway at the startup
+
+
+  [![]({{base_path}}/assets/img/learn/gateway-startup.png)]({{base_path}}/assets/img/learn/gateway-startup.png)
+  
+At startup, the gateway will look on the APIs with labels which it is subscribed to, in the extension, and fetch the synapse artifacts of those APIs. Those synapse artifacts will get deployed in the gateway.
+
+Gateways are subscribed to the traffic manager. There is an extension in the gateway to get the synapse artifacts and deploy them in the memory. Gateways can subscribe to multiple labels.
+
+##Configuration Related to Artifact Synchronizer
+
+You need to configure Gateway and Publisher node as given below to save artifacts and retrieve artifacts through
+corresponding extensions
+
+###Publisher Profile 
+
+Update `<API-M_HOME>/repository/conf/deployment.toml` file as follows, and change the default values according to your
+ need. 
+
+```
+[apim.sync_runtime_artifacts.publisher]
+artifact_saver = "DBSaver"
+publish_directly_to_gateway = "true"
+```
+
+ - Through artifact_saver we can specify the extension point. The default is `DBSaver` where the artifacts are saved in
+ the database.
+ - If `publish_directly_to_gateway = true` then the artifacts will be published to the gateway directly. If
+ `publish_directly_to_gateway = false` then the published API details will be notified to TM through events.
+ - If we have `[apim.sync_runtime_artifacts.publisher]` configuration element then all the artifacts will be saved to
+  the storage via the extension point. In default they will get stored in Database.
+  
+  We can add the gateways as environments. For more information see 
+   [Adding gateways as Environments]({{base_path}}/learn/api-gateway/maintaining-separate-production-and-sandbox-gateways/).
+
+###Gateway Profile Configuration
+
+Update `<API-M_HOME>/repository/conf/deployment.toml` file as follows, and change the default values according to your
```

Comment:
- Open the `<API-M_HOME>/repository/conf/deployment.toml` file and add the following configuraton.
```
[apim.sync_runtime_artifacts.gateway]
gateway_labels =["Production and Sandbox","Label1","Label2"]
artifact_retriever = "DBRetriever"
deployment_retry_duartion = 15000
data_retrieval_mode = "sync"
save_artifacts_locally = false
```
| Configuration| Description|
|-------------------|----------------|
| `gateway_labels` | the labels which the gateway is subscribed to. Only the APIs with these  labels will be pulled from the extension point and deployed.|
|`artifact_retriever ` |  The configured extension point to retrieve artifacts from. The default is `DBRetriever` where the artifacts are retrieved from the database.|
|`deployment_retry_duartion` |  the retry duration in milliseconds to deploy artifacts if there is a failure in retrieving from the extension. The retry duration specified here will be exponentially increased by a progression factor of 2. For example, if there are conitnuous failures the retry duration is progressed as 15s, 30s, 60s, 120s,... .The retry duration is bounded by 1 hr.
    !!! note 
        This `deployment_retry_duartion` is  applicable only in Asynchronous deployment ( `data_retrieval_mode = "async"` )  where the server will try to deploy the artifacts after it is started. |
| `data_retrieval_mode = “sync” ` |  the mode of deployment of artifacts from the exten                       sion point. By default, the gateway startup will occur in synchronous mode where the server will wait until all the APIs have been deployed. If there are any failures in the deployment process, the deployment will be triggered for `n` times where `n` is the maximum retry count configured. If the APIs are not deployed even in `n` retries then the server will start with un-deployed API artifacts.
To startup the gateway in asynchronous mode  the configuration should be specified as   `data_retrieval_mode = "async"`. Here the server will be up, irrespective of the deployment status of the synapse artifacts in the gateway.|
|`save_artifacts_locally = true` | adding this config or removing `save_artifacts_locally` config  will store synaps artifacts in the file system. (Saved in
   `<APIM_HOME>/repository/deployment/server/synapse-configs/default/`) . When `save_artifacts_locally = false`  then the artifacts from extension point will not be stored in the file system.|